Playmate/View Controllers/Session Details View/SessionDetailsViewController.m

```diff
@@ -161,27 +145,94 @@ - (void)stopConfetti {
 #pragma mark - Add to session button action
 
 - (IBAction)addMyself:(id)sender {
-    [self showConfetti];
-    
-    PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
-
-    // Retrieve the object by id
-    [query getObjectInBackgroundWithId:self.sessionDetails.objectId
-                                 block:^(PFObject *session, NSError *error) {
-        
-        // Add myself to the session
-        NSMutableArray *oldPlayersList = (NSMutableArray *)session[@"playersList"];
-        [oldPlayersList addObject:me];
-        
-        session[@"playersList"] = (NSArray *)oldPlayersList;
+    // For leaving session
+    if (partOfSession) {
+        [self updateLeaveUi];
+        [self changeAddButtonToJoin];
         
-        int oldOccupied = [session[@"occupied"] intValue] + 1;
-        session[@"occupied"] = [NSNumber numberWithInt:oldOccupied];
-        
-        [session saveInBackground];
+        PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
+
+        // Retrieve the object by id
+        [query getObjectInBackgroundWithId:self.sessionDetails.objectId
+                                     block:^(PFObject *session, NSError *error) {
+            // Add myself to the session
+            NSMutableArray *oldPlayersList = (NSMutableArray *)session[@"playersList"];
+            
+            PFUser * _Nullable userToRemove = nil;
+            for (PFUser *user in oldPlayersList) {
+                if ([user.objectId isEqualToString:me.objectId]) {
+                    userToRemove = user;
+                    break;
+                }
+            }
+            
+            [oldPlayersList removeObject:userToRemove];
+                        
+            session[@"playersList"] = (NSArray *)oldPlayersList;
+            
+            int newOccupied = (int)oldPlayersList.count;
```

Comment:
do you need to cast to int?

Reply:
array.count returns a long so I use this to get rid of warning. Hopefully there is not over 2147483647 people participating in one session :)